bottlecap/src/logs/flusher.rs

```diff
@@ -93,14 +96,18 @@ impl Flusher {
         failed_requests
     }
 
-    async fn create_request(&self, data: Vec<u8>) -> reqwest::RequestBuilder {
+    async fn create_request(&self, data: Vec<u8>) -> Option<reqwest::RequestBuilder> {
         let url = format!("{}/api/v2/logs", self.endpoint);
-        let headers = self.get_headers().await;
-        self.client
+        let Some(headers) = self.get_headers().await else {
+            return None;
```

Comment:
Mmm, although I understand the code, I find it a little confusing that `get_headers` is responsible for deciding wether or not we're creating a request.

Would it make more sense to rearchitect this so that whenever we definitely know we are about to flush, let's say in `flush()` method, we try to get the API Key?